lib/thread_local.rb

```diff
@@ -101,17 +101,18 @@ def forget_vanished_threads!
     end
   end
 
-  def define_initializer!
-    ObjectSpace.define_finalizer self do
```

Comment:
Looking into code, I've realized that the proc created via this `do ... end` block referenced `self`.
Thus `self` is never GCed/finalized.
Really a good catch!